In [1]:
%pylab inline
import pandas as pd
from pandas import Series, DataFrame

s = pd.read_csv('data/taxi-distance.csv', squeeze=True, header=None)

Populating the interactive namespace from numpy and matplotlib


# Beyond 1

Compare the mean and median trip distances. What does that tell you about the distribution of our data?

In [2]:
s.describe()

count    9999.000000
mean        3.158511
std         4.037516
min         0.000000
25%         1.000000
50%         1.700000
75%         3.300000
max        64.600000
Name: 0, dtype: float64

Because the mean is significantly higher than the median, it would seem that there are some *very* long trips in our data set that are pulling the mean up. And sure enough, we see that the standard deviation is 4, but that we have at least one trip > 64 miles in length.

# Beyond 2

How many short, medium, and long trips were there for trips that had only one passenger? Note that data for passenger count and trip length are from the same data set, meaning that the indexes are the same.

In [3]:
passenger_count = pd.read_csv('data/taxi-passenger-count.csv', squeeze=True, header=None)

pd.cut(s[passenger_count == 1], 
       bins=[s.min(), 2, 10, s.max()], 
       labels=['short', 'medium', 'long']).value_counts()

short     4285
medium    2387
long       487
Name: 0, dtype: int64

# Beyond 3

What happens if we don't pass explicit intervals, and instead ask `pd.cut` to just create 3 bins, with `bins=3`?

In [4]:
passenger_count = pd.read_csv('data/taxi-passenger-count.csv', squeeze=True, header=None)

pd.cut(s[passenger_count == 1], 
       bins=3,
       labels=['short', 'medium', 'long'], retbins=True)

(0       short
 1       short
 2       short
 3       short
 4       short
         ...  
 9993    short
 9994    short
 9995    short
 9996    short
 9998    short
 Name: 0, Length: 7207, dtype: category
 Categories (3, object): ['short' < 'medium' < 'long'],
 array([-0.0646    , 21.53333333, 43.06666667, 64.6       ]))

`pd.cut` took the interval from `s.min()` to `s.max()`, divided it into three equal parts, and assigned those to be `short`, `medium`, and `long`. We can see, though, that this meant our `long` category is from 43 miles to 64.6 miles -- numerically one-third of the values' interval, but only including a handful of values!